warehouse/query-core/src/test/java/datawave/query/iterator/logic/TermFrequencyIndexIteratorTest.java

```diff
@@ -337,23 +349,132 @@ public void testScanFullRangeExclusiveEventDataQueryExpressionFilter() throws IO
         // jump to the first doc
         iterator.seek(null, null, true);
         
-        Assert.assertTrue(iterator.hasTop());
+        assertTrue(iterator.hasTop());
         Document d = iterator.document();
-        Assert.assertTrue(d != null);
-        Assert.assertTrue(d.getDictionary().size() == 2);
-        Assert.assertTrue(d.getDictionary().get("FOO") != null);
-        Assert.assertTrue(d.getDictionary().get("RECORD_ID") != null);
-        Assert.assertTrue(d.getDictionary().get("FOO").getData() != null);
+        assertTrue(d != null);
+        assertTrue(d.getDictionary().size() == 2);
+        assertTrue(d.getDictionary().get("FOO") != null);
+        assertTrue(d.getDictionary().get("RECORD_ID") != null);
+        assertTrue(d.getDictionary().get("FOO").getData() != null);
         Iterator<PreNormalizedAttribute> i = ((Set) d.getDictionary().get("FOO").getData()).iterator();
-        Assert.assertTrue(i.next().getValue().equals("bar"));
-        Assert.assertTrue(i.next().getValue().equals("baz"));
-        Assert.assertTrue(i.next().getValue().equals("buf"));
-        Assert.assertTrue(i.next().getValue().equals("buz"));
-        Assert.assertTrue(i.next().getValue().equals("alf"));
-        Assert.assertTrue(i.next().getValue().equals("arm"));
+        assertTrue(i.next().getValue().equals("bar"));
+        assertTrue(i.next().getValue().equals("baz"));
+        assertTrue(i.next().getValue().equals("buf"));
+        assertTrue(i.next().getValue().equals("buz"));
+        assertTrue(i.next().getValue().equals("alf"));
+        assertTrue(i.next().getValue().equals("arm"));
         
         iterator.next();
-        Assert.assertFalse(iterator.hasTop());
+        assertFalse(iterator.hasTop());
+    }
+    
+    @Test
+    public void testZeroItem() {
+        SortedMapIterator source = getIterator();
+        
+        Key start = new Key("row", "fi\0FIELD", "value\u0000datatype\u00001");
+        Key end = new Key("row", "fi\0FIELD", "value\uffff\u0000datatype\u00001\uffff");
+        
+        TermFrequencyIndexIterator tfIter = new TermFrequencyIndexIterator(start, end, source, TimeFilter.alwaysTrue());
+        IndexIteratorBridge iter = new IndexIteratorBridge(tfIter, JexlNodeFactory.buildEQNode("FIELD", "value"), "FIELD");
+        iter.seek(new Range(), Collections.emptyList(), false);
+        
+        int i = 0;
+        while (iter.hasNext()) {
+            Key key = iter.next();
+            assertEquals("received " + key.toStringNoTime(), "datatype\u00001", key.getColumnFamily().toString());
+            i++;
+        }
+        
+        assertEquals("should have one entry but got " + i, 1, i);
```

Comment:
Following up on our conversation - shouldn't this also return no entries because there is no "FIELD=value" entry in the source TF rows?  It looks like its just returning the first entry in source after the start key

Reply:
This is aggregating all matching uids into a `datatype\0uid`, so for the given dataset we can expect one entry only.